src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -117,15 +122,39 @@ public async Task StopAsync(CancellationToken cancellationToken)
             }
         }
 
-        private void InitializeHost()
+        internal virtual async Task StartProcessorAsync()
         {
             if (this._host == null)
             {
-                this._host = new ChangeFeedEventHost(this._hostName,
-                    this._monitorCollection,
-                    this._leaseCollection,
-                    this._changeFeedOptions,
-                    this._leaseHostOptions);
+                this._host = await this._hostBuilder.BuildAsync().ConfigureAwait(false);
+            }
+
+            await this._host.StartAsync().ConfigureAwait(false);
+        }
+
+        internal virtual void WithHealthMonitor(IHealthMonitor healthMonitor)
+        {
+            this._healthMonitor = healthMonitor ?? throw new ArgumentNullException(nameof(healthMonitor));
+        }
+
+        private void InitializeBuilder()
+        {
+            if (this._hostBuilder == null)
+            {
+                if (this._healthMonitor == null)
+                {
+                    this._healthMonitor = new CosmosDBTriggerHealthMonitor(this._logger);
+                }
```

Comment:
Can you initialize this in the constructor?

Reply:
Done